### ISSUE TITLE
Handle "release" builds

### ISSUE THREAD

**Issue by lazyfrosch** [Icinga/puppet-icinga_build#14]
Distinguish between "release" and "snapshot" builds.

## Release

Build the software in the version that is specified inside `debian/changelog` or `<product>.spec`.

We can use `uscan` for Debian and `spectool` for RPM to get the correct upstream tarball.

## Snapshot

Get a current tarball for a branch of the upstream software, so we can build a on demand package.

**Comment by lazyfrosch** [comment]
This should be controlled by parameters:

``` puppet
icinga_build::pipeline { 'icinga2':
  product => 'icinga2',
  type    => 'release', # type release/snapshot
  target  => '', # repo suffix
  # ...
}

icinga_build::pipeline { 'icinga2-snapshot':
  product => 'icinga2',
  type    => 'snapshot',
  target  => '-snapshots', # repo suffix (default value per type?)
  # ...
}
```

This will then control:

* How upstream tarball is retrieved
  * Snapshot script
  * Or spectool / uscan download
* How version is calculated (basically only for snapshots)
  * Release builds will use the version of `.spec` or `debian/changelog`
* What repository the package will get pushed to
  * Maybe disable publish when target is false? (absent the publish job?)
* Publish jobs in mode release should only publish manually
  * Maybe fail the job when a parameter on the publish job isn't set to true
    (or warning)
  * Only auto-publish in snapshot type

**Comment by lazyfrosch** [comment]
## Debian

We won't have to modify changelog in release mode, the tarball can be downloaded via uscan:

    cd icinga2/jessie
    uscan --download-current-version --verbose

Will download the files and provide the proper tarball name ready to use:

    ls -l ../*.tar*
    lrwxrwxrwx 1 markus markus      20 Mär  1 10:55 ../icinga2_2.6.0.orig.tar.gz -> icinga2-2.6.0.tar.gz
    -rw-r--r-- 1 markus markus 2353930 Mär  1 10:55 ../icinga2-2.6.0.tar.gz

**Comment by lazyfrosch** [comment]
## RPM

When Source is a valid URL inside the RPM spec `spectool` can download the tarball. (Which is recommended)

The spec file shouldn't be modified then, version, revision and changelog are controlled by `icinga-packaging`

    $ cd icinga2/
    $ spectool -g icinga2.spec
    Getting https://github.com/Icinga/icinga2/archive/v2.6.0.tar.gz to ./v2.6.0.tar.gz
    $ ls -l
    -rw-r--r-- 1 build build   23197 Mar  1 09:58 icinga2.spec
    -rw-r--r-- 1 root  root  2353930 Mar  1 09:59 v2.6.0.tar.gz

**Comment by Crunsher** [comment]
There exist a dedicated release pipeline now. This issue can be closed once we figure out #20
